scripts/azureml-assets/azureml/assets/update_assets.py

```diff
@@ -63,6 +65,34 @@ def release_tag_exists(asset_config: assets.AssetConfig, release_directory_root:
     return asset_config.full_name in repo.tags
 
 
+
+def validate_new_release(asset_config: assets.AssetConfig, release_directory_root: Path) -> bool:
+    """Check repo to see if an asset's previous version was released if a latest tag exists.
+
+    Args:
+        asset_config (assets.AssetConfig): Asset config
+        release_directory_root (Path): Release branch location
+
+    Returns:
+        bool: True if the previous tag exists, False otherwise
+    """
+    if asset_config.type != AssetType.ENVIRONMENT:
+        return True
+
+    repo = Repo(release_directory_root)
+    tags = [t for t in repo.tags if tag.name.startswith(f"{asset_config.partial_name}/")]
+    if not tags:
+       # No releases
+        return True
+
+    # Get the latest tag
+    ordered_tags = sorted(tags, key=lambda t: t.tag.tagged_date)

```

Comment:
For just the TagReference (t) environment/promptflow-runtime/1, the TagObject (t.tag) is None, which causes an exception to be thrown when trying to access t.tag.tagged_date

Reply:
Thanks for testing this! I think that tag must be lightweight instead of annotated ([reference](https://git-scm.com/book/en/v2/Git-Basics-Tagging#:~:text=list%20is%20mandatory.-,Creating%20Tags,-Git%20supports%20two)). Rather than making this super complicated, maybe we can go with the lowest common denominator (commit date) after all, assuming that nobody is going to tag something out of sequence?

```suggestion
    ordered_tags = sorted(tags, key=lambda t: t.commit.authored_datetime)
```

If we were to try using the tag timestamp first and then falling back to the commit date as needed, we'd have to do a lot of date conversions that wouldn't be fun, since the former is stored as seconds since the epoch and the latter as a datetime object. AND let's say we were to change that tag to an annotated one...it would have a newer timestamp than all of the other tags in the repo which would throw the sort off.

Reply:
Got it, thanks for the explanation!